src/cpp/lpnamer/input_reader/LpFilesExtractor.cpp

```diff
@@ -1,14 +1,41 @@
 #include "LpFilesExtractor.h"
 
+#include <algorithm>
 #include <sstream>
 
 #include "ArchiveReader.h"
 
 void LpFilesExtractor::ExtractFiles() const {
-  auto archive_reader = ArchiveReader(antares_archive_path_);
-  archive_reader.Open();
-  const auto vect_area_files =
-      archive_reader.ExtractPattern("area*.txt", "", xpansion_output_dir_);
+  std::vector<std::filesystem::path> vect_area_files;
+  std::vector<std::filesystem::path> vect_interco_files;
+
+  // TODO: test unknown
+  if (mode_ == Mode::ARCHIVE) {
+    auto archive_reader = ArchiveReader(antares_archive_path_);
+    archive_reader.Open();
+    vect_area_files =
+        archive_reader.ExtractPattern("area*.txt", "", xpansion_output_dir_);
+
+    vect_interco_files =
+        archive_reader.ExtractPattern("interco*.txt", "", xpansion_output_dir_);
+
+    archive_reader.ExtractPattern("ts-numbers*", "", xpansion_output_dir_);
+    archive_reader.Close();
+    archive_reader.Delete();
+  } else {
```

Comment:
Why not be more explicit and putting `else if Mode::ANTARES_API || Mode::FILE`  ? As there is also an unknown mode (but error would have been raised before ?)

Reply:
Indeed.

Now properly supported and future prof with default case